### DISCUSSION TITLE
Use Shipping address as backup for Standard Tax calculation

### DISCUSSION THREAD

**Discussion by Web10-Joris** [duncanmcclean/simple-commerce#1023] [Feature Requests]
Hi Duncan,

I've been debugging my environment and came to this case where a Prevent Checkout exception was thrown. 

Details: 
1. Use Standard Tax Engine & Database orders
2. Set `simple-commerce.tax_engine_config.address` to 'billing'
3. Set `simple-commerce.tax_engine_config.behaviour.no_rate_available` to  'prevent_checkout'
4. Create a order with only shipping address data and `use_shipping_address_for_billing` to 0 
5. The order fails checkout (because of the prevent_checkout setting). 

Is there a way to use the shipping address as backup for calculating the Tax when no billing address is filled in? 
Or a condition to check if there is there is a Billing Address for the order when updating it, if not, set `use_shipping_address_for_billing` to true.

**Comment by duncanmcclean** [comment]
Simple Commerce _should_ already fallback to the shipping address if you were to set `use_shipping_address_for_billing` to `true`. Do you still get that error after changing the setting?

**Reply by Web10-Joris** [reply]
Indeed, my bad. I thought it was maybe nice to have this option to be `true` by default. Now emtpy (new) carts without this form input will have `false`. But this can be fixed per environment by adding (hidden) `use_shipping_address_for_billing`  field with checked value as well.
